### PR TITLE
up to date abi with updated 875 spec

### PR DESCRIPTION
**Pull request by James-Sangalli**
for #812

### CODE REVIEW

File: AlphaWallet/RPC/Commands/web3wift-web3js/ContractERC875Transfer.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@ struct ContractERC875Transfer: Web3Request {
     let indices: [UInt16]
     
     var type: Web3RequestType {
-        let abi = "{\"constant\":false,\"inputs\":[{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"ticketIndices\",\"type\":\"uint16[]\"}],\"name\":\"transfer\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}, [\"\(address)\", \(indices)]"
+        let abi = "{\"constant\":false,\"inputs\":[{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"ticketIndices\",\"type\":\"uint256[]\"}],\"name\":\"transfer\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}, [\"\(address)\", \(indices)]"
```
Comment (hboon) [inline AlphaWallet/RPC/Commands/web3wift-web3js/ContractERC875Transfer.swift]:
```suggestion
        let abi = "{\"constant\":false,\"inputs\":[{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"indices\",\"type\":\"uint256[]\"}],\"name\":\"transfer\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}, [\"\(address)\", \(indices)]"
```

File: AlphaWallet/RPC/Commands/web3wift-web3js/ClaimERC875Order.swift [hunk 2]
```diff
@@ -29,7 +29,7 @@ struct ClaimERC875Order: Web3Request {
     let s: String
 
     var type: Web3RequestType {
-        let abi = "{\"constant\":false,\"inputs\":[{\"name\":\"expiry\",\"type\":\"uint256\"},{\"name\":\"ticketIndices\",\"type\":\"uint16[]\"},{\"name\":\"v\",\"type\":\"uint8\"},{\"name\":\"r\",\"type\":\"uint256\"},{\"name\":\"s\",\"type\":\"uint256\"}],\"name\":\"trade\",\"outputs\":[],\"payable\":true,\"stateMutability\":\"payable\",\"type\":\"function\"}, [\"\(expiry)\", \(indices), \(v), \"\(r)\", \"\(s)\"]"
+        let abi = "{\"constant\":false,\"inputs\":[{\"name\":\"expiry\",\"type\":\"uint256\"},{\"name\":\"indices\",\"type\":\"uint256[]\"},{\"name\":\"v\",\"type\":\"uint8\"},{\"name\":\"r\",\"type\":\"uint256\"},{\"name\":\"s\",\"type\":\"uint256\"}],\"name\":\"trade\",\"outputs\":[],\"payable\":true,\"stateMutability\":\"payable\",\"type\":\"function\"}, [\"\(expiry)\", \(indices), \(v), \"\(r)\", \"\(s)\"]"
```
Comment (hboon) [inline AlphaWallet/RPC/Commands/web3wift-web3js/ClaimERC875Order.swift]: Do we want to take the chance and change `r`, `s` to `bytes32`?

**Comment by James-Sangalli** [comment]
@hboon dependant on: https://github.com/AlphaWallet/alpha-wallet-ios/issues/842

**Comment by James-Sangalli** [comment]
@hboon this is ready. It works with both legacy contracts and new ones, however our indices inside swift are still UInt16's meaning people can still use the new spec with our app, they are just limited to the size of 16 bits.

I kept the UInt16's due to the special encoding format which can allow two tokens to be encoded in one index. I can create a seperate PR to still allow this encoding if the BigUInt is below the 16 bit threshold.

Also fixes #842

**Comment by James-Sangalli** [comment]
@JamesSmartCell need this implemented in android soon too as the spec will be upgraded for the factory contract builder

**Comment by James-Sangalli** [comment]
@hboon only these two matter for now, can add later. 

See: getTokenIndicesFromLinkBytes in UniversalLinkHandler to understand what I mean

**Pull request merged**